app/process/watcher/evm/watcher.go

```diff
@@ -85,58 +91,95 @@ func NewWatcher(
 		log.Tracef("[%s] - Updated Transfer Watcher timestamp to [%s]", contracts.Address(), timestamp.ToHumanReadable(startBlock))
 	}
 	return &Watcher{
-		repository:  repository,
-		contracts:   contracts,
-		evmClient:   evmClient,
-		logger:      c.GetLoggerFor(fmt.Sprintf("EVM Router Watcher [%s]", contracts.Address())),
-		mappings:    mappings,
-		targetBlock: targetBlock,
-		validator:   validator,
+		repository:    repository,
+		contracts:     contracts,
+		evmClient:     evmClient,
+		logger:        c.GetLoggerFor(fmt.Sprintf("EVM Router Watcher [%s]", contracts.Address())),
+		mappings:      mappings,
+		targetBlock:   targetBlock,
+		validator:     validator,
+		sleepDuration: sleepDuration,
 	}
 }
 
 func (ew *Watcher) Watch(queue qi.Queue) {
-	go ew.listenForEvents(queue)
+	go ew.beginWatching(queue)
 
-	ew.processPastLogs(queue)
 	ew.logger.Infof("Listening for events at contract [%s]", ew.contracts.Address())
 }
 
-func (ew Watcher) processPastLogs(queue qi.Queue) {
+func (ew Watcher) beginWatching(queue qi.Queue) {
 	fromBlock, err := ew.repository.Get(ew.contracts.Address().String())
 	if err != nil {
 		ew.logger.Errorf("Failed to retrieve EVM Watcher Status fromBlock. Error: [%s]", err)
-		return
 	}
 
 	ew.logger.Infof("Processing events from [%d]", fromBlock)
 
+	for {
+		fromBlock, err := ew.repository.Get(ew.contracts.Address().String())
+		if err != nil {
+			ew.logger.Errorf("Failed to retrieve EVM Watcher Status fromBlock. Error: [%s]", err)
+			continue
+		}
+
+		currentBlock, err := ew.evmClient.BlockNumber(context.Background())
+		if err != nil {
+			ew.logger.Errorf("Failed to retrieve latest block number. Error [%s]", err)
+			time.Sleep(ew.sleepDuration)
+			continue
+		}
+
+		toBlock := int64(currentBlock - ew.evmClient.BlockConfirmations())
+		if fromBlock > toBlock {
+			time.Sleep(ew.sleepDuration)
+			continue
+		}
+
+		err = ew.processLogs(fromBlock, toBlock, queue)
+		if err != nil {
+			ew.logger.Errorf("Failed to process logs. Error: [%s].", err)
+			time.Sleep(ew.sleepDuration)
+			continue
+		}
+
+		time.Sleep(ew.sleepDuration)
+	}
+}
+
+func (ew Watcher) processLogs(fromBlock, endBlock int64, queue qi.Queue) error {
 	abi, err := abi.JSON(strings.NewReader(router.RouterABI))
 	if err != nil {
 		ew.logger.Errorf("Failed to parse router ABI. Error: [%s]", err)
-		return
+		return err
 	}
+
 	burnHash := abi.Events["Burn"].ID
 	lockHash := abi.Events["Lock"].ID
+	memberUpdatedHash := abi.Events["MemberUpdated"].ID
 
 	topics := [][]common.Hash{
 		{
 			burnHash,
 			lockHash,
+			memberUpdatedHash,
 		},
 	}
 	addresses := []common.Address{
 		ew.contracts.Address(),
```

Comment:
Are we 100% sure that events with the same signature, emitted in a `router` TX will be ignored

Reply:
That's why you are filtering by `addresses`. They are used to restrict matches to events created by those specific addresses.